pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1369,24 +1395,24 @@ func (c *PodAffinityChecker) satisfiesExistingPodsAntiAffinity(pod *v1.Pod, meta
 			glog.Error(errMessage)
 			return ErrExistingPodsAntiAffinityRulesNotMatch, errors.New(errMessage)
 		}
-		if matchingTerms, err = c.getMatchingAntiAffinityTerms(pod, filteredPods); err != nil {
+		if _, topologyPairToMatchingPods, err = c.getMatchingAntiAffinityTopologyPairs(pod, filteredPods); err != nil {
 			errMessage := fmt.Sprintf("Failed to get all terms that pod %+v matches, err: %+v", podName(pod), err)
 			glog.Error(errMessage)
 			return ErrExistingPodsAntiAffinityRulesNotMatch, errors.New(errMessage)
 		}
 	}
-	for _, terms := range matchingTerms {
-		for i := range terms {
-			term := &terms[i]
-			if len(term.term.TopologyKey) == 0 {
-				errMessage := fmt.Sprintf("Empty topologyKey is not allowed except for PreferredDuringScheduling pod anti-affinity")
-				glog.Error(errMessage)
-				return ErrExistingPodsAntiAffinityRulesNotMatch, errors.New(errMessage)
-			}
-			if priorityutil.NodesHaveSameTopologyKey(node, term.node, term.term.TopologyKey) {
-				glog.V(10).Infof("Cannot schedule pod %+v onto node %v,because of PodAntiAffinityTerm %v",
-					podName(pod), node.Name, term.term)
-				return ErrExistingPodsAntiAffinityRulesNotMatch, nil
+
+	// Iterate over topology topology pairs to get any of the pods being affected by
+	// the scheduled pod anti-affinity rules
+	for topologyKey, topologyValue := range node.Labels {
+		if violatedPods, ok := topologyPairToMatchingPods[topologyPair{key: topologyKey, value: topologyValue}]; ok {
+			affinity := violatedPods[0].Spec.Affinity
+			for _, term := range GetPodAntiAffinityTerms(affinity.PodAntiAffinity) {
```

Comment:
I don't think you need to check the terms here.

Reply:
That was just to provide the term violated in the error message. Do you think that we need the error to be that verbose? or can we ignore it?

Reply:
I see. You are not returning the error message to the user in the new code though. I think it is useful to return a better error message to the user.

Reply:
When I looked back at the code, I realized that returning an error here, does not return a better error message to the user. I just indicates that an error has occurred during the execution of the predicate. We shouldn't return the error and we should not waste cycles to iterate over the terms only for logging at V(10).